### PR TITLE
Feature-3617 Defining Resource Limit for Pods

### PR DESCRIPTION
**Pull request by mishah334**
## Description

Defining Resource Limit for Pods.

## Related Issues

[#3617](https://github.com/astronomer/issues/issues/3617)

## Testing

Added the test case to check resource limit definations.

## Merging

N/A

### CODE REVIEW

File: charts/stan/values.yaml [hunk 1]
```diff
@@ -161,13 +165,14 @@ store:
...
-  #   memory: 128Mi
-  #  requests:
-  #   cpu: 100m
-  #   memory: 128Mi
+  resources:
+    limits:
+     cpu: 100m
+     memory: 100Mi
+    requests:
+     cpu: 100m
```
Comment (santoshr1016) [inline charts/stan/values.yaml:173]: Requests: you can change it to CPU 50m, memory to 80Mi

File: charts/nats/values.yaml [hunk 2]
```diff
@@ -188,13 +188,13 @@ reloader:
...
-  #   memory: 128Mi
-  #  requests:
-  #   cpu: 100m
-  #   memory: 128Mi
+  resources:
+    limits:
+     cpu: 100m
+     memory: 128Mi
+    requests:
+     cpu: 100m
```
Comment (santoshr1016) [inline charts/nats/values.yaml:196]: Requests: CPU to 50m. memory to 100Mi
idea is not to be same as limits

**Review by santoshr1016 — APPROVED** [review]
+1

File: charts/stan/values.yaml (hunk 1 above)
Comment (pgvishnuram) [reply charts/stan/values.yaml:173]: https://github.com/astronomer/astronomer/blob/master/values.yaml#L401-L413

File: charts/nats/values.yaml (hunk 2 above)
Comment (pgvishnuram) [reply charts/nats/values.yaml:196]: https://github.com/astronomer/astronomer/blob/master/values.yaml#L387-L399

**Review by danielhoherd — APPROVED** [review]
Looks good!

**Pull request merged**